### DISCUSSION TITLE
Prevent Language Subfolders

### DISCUSSION THREAD

**Discussion by ShareBuilderAustin** [dahall/TaskScheduler#978] [Q&A] [answered]
Is there a way to prevent the Language Localization Subfolders when compiling in .net Framework 4.6.2? (VS community 2022)

my program only needs to worry about english, 
I tried adding "\<SatelliteResourceLanguages\>en\</SatelliteResourceLanguages\>" without success

the compiler still creates de, es, fr, it, pl, ru, zh-cn, zh-hant subfolders in bin\ when copylocal is true on the Task Scheduler Managed Wrapper reference

Thank you for a great library!

**Comment by dahall** [comment, marked as answer]
I have never found a way with project settings or msbuild options. However, you can use a post-build event that deletes all the subfolders.

**Comment by ShareBuilderAustin** [comment]
Thank you David, that’s a good idea,
The workaround I used was to delete the subfolders in the nuget package reference folder.

**Comment by facreationz** [comment]
Is it safe to delete the language subfolders or exclude them when compiling a distributable setup? 

Also, how are these languages brought into use if they are needed? Meaning, are these languages auto-selected by the library depending upon the user's regional settings or do I have to manually specify which language to use?

Thanks for the excellent library, its has definitely made scheduled task management an easy job.

**Reply by dahall** [reply]
> Is it safe to delete the language subfolders or exclude them when compiling a distributable setup?

Yes 
> Also, how are these languages brought into use if they are needed? Meaning, are these languages auto-selected by the library depending upon the user's regional settings or do I have to manually specify which language to use?

The .NET framework automatically searches for and uses the current user local when launching an assembly. The developer does not need to do anything other than [enable globalization in their code](https://learn.microsoft.com/en-us/dotnet/core/extensions/best-practices-for-developing-world-ready-apps).
> Thanks for the excellent library, its has definitely made scheduled task management an easy job.

You're very welcome. I'm glad it is useful.
